src/script/descriptor.cpp

```diff
@@ -1040,6 +1156,14 @@ class MiniscriptDescriptor final : public DescriptorImpl
 
     bool IsSolvable() const override { return true; }
     bool IsSingleType() const final { return true; }
+
+    std::optional<int64_t> ScriptSize() const override { return m_node->ScriptSize(); }
+
+    std::optional<int64_t> MaxSatSize(bool) const override {
+        // We only ever parse satisfiable Miniscripts. Also note for Miniscript we always assume
+        // high-R ECDSA signatures.
+        return CHECK_NONFATAL(m_node->GetWitnessSize());
```

Comment:
This is not true if the Miniscript descriptor was parsed from Script. I'll update this.

Reply:
Done.